### DISCUSSION TITLE
Docker service?

### DISCUSSION THREAD

**Discussion by dmikushin** [yuezk/GlobalProtect-openconnect#215] [General]
Hello, thank you for the great project! It really helps many of us to survive within the corporate ~~stupidness~~ security.

One more step forward: I'd really wish to lock this thing down into a Docker container and run somewhere in my network. Limitation is that like the original client, this project relies on DBus and Systemd, which are generally not available within a Docker container. I tried to run dbus manually for PulseSecure, a similar VPN shitware, without success. Could you kindly offer and advise on how to run your project in a bare Docker environement, e.g. within a single bash script?

**Comment by yuezk** [comment]
I’m planning to rewrite this client and it will not depend on DBus. And can be deployed in Docker, before that I’m afraid I don’t have time to invest in it.

**Reply by dmikushin** [reply]
Hi @yuezk , here is your client bundled into a Docker container: https://github.com/dmikushin/globalprotect-docker I hope it's fine to reference a fork here, and I thank everyone once again for the hard work it is build on top of.

The fastest solution, not involving much code changes appears to be just to couple client and service into a single executable. Then there is no need for D-Bus anymore, but instead there is a need for admin rights. (Whoever has time, could possibly implement client-service with gRPC.) Qt apps have a builtin VNC server, which we can expose to the web browser via noVNC. One option to share the containerized VPN connectivity is to expose it as a SOCKS5 proxy. In the end of the day, we have everything running in the container, and SOCKS5 is provided by SSH.
